Automattic/jetpack
# VideoPress: Expire playback tokens

Fixes #27070.

This is part of an effort to reuse playback tokens as much as possible. For now we keep them on the wp.data redux state, but we want to make them survive to page refreshes by storing them on local storage/ indexedDB. Being able to expire tokens is a requirement for this strategy to work as expected.

#### Changes proposed in this Pull Request:

* Create an action to remove expired tokens from state
* Trigger the new action when a requested token had expired

#### Other information:


#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* You will need to test it on a JN site (* see notes)
* Keep the browser debug console open
* Go to `Jetpack > VideoPress` to access your library
* Upload at least 7 videos to see more than 1 page of videos
* Update 1 video privacy to private:

![image](https://user-images.githubusercontent.com/6760046/198618461-d911f344-abf2-461f-a474-4fb78e9ed026.png)

* On the console, check the `Network` tab for a playback token request:

![image](https://user-images.githubusercontent.com/6760046/198621217-f35806dc-3b19-4764-8718-b06e957d578c.png)

* The requested token will be expired by the client code after 24h, so we need to make the time pass: update your computer's date to +1 day in the future
* On your `VideoPress library`, navigate between pages (that's why you need 7 or more uploads) to trigger some video loading
* On the console, check the `Network` tab for a new playback token request
* Verify that the thumbnail for the private video is shown correctly
* You can advance your date yet another day and try navigating between pages again, to trigger another token request

Notes: 

* you may need to test it on a JN site because setting a new date on your computer will affect any local wordpress installation as well, and the WPCOM API request to get new tokens will break due to the date mismatch between you and the WPCOM server; using a JN site will ensure the backend will be on the same date while we simulate the time passing on the client side
